hpeOneView/resources/settings/appliance_proxy_configuration.py

```diff
@@ -39,3 +39,24 @@ def __init__(self, connection, data=None):
 
     def get_all(self):
         return super(ApplianceProxyConfiguration, self).get_by_uri(self.URI)
+
+    def get_by_proxy(self, proxy_ip):
+        """Retrieves a resource by proxy server ip.
+
+        Args:
+            proxy_ip: Ip address of the proxy
+
+        Returns:
+            Resource object or None if resource does not exist.
+        """
+        results = self.get_all().data
+        if results:
+            if str(results.get("server", "")).lower() == proxy_ip.lower():
+                new_resource = self.new(self._connection, results)
+            else:
+                new_resource = None
+        return new_resource
+
+    def delete(self):
```

Comment:
no need to add this method separately. Existing ```delete``` method in resource class will pick the uri from self.data and will perform delete operation.
Any specific reason to add this method..? Also please add UTs.

Reply:
But we dont have uri field in self.data.its None.so here, Im updating self.data['uri'] = self.URI.

Reply:
ok. In that case I suggest you not to add ```uri``` field in self.data and try using ```ResourceHelper``` class where you can pass ```uri``` as a param as shown below.
```return self._helper.delete(self.URI, timeout=timeout, force=force)```

Reply:
Done